src/careamics/dataset/iterable_dataset.py

```diff
@@ -91,38 +91,36 @@ class PathIterableDataset(IterableDataset):
     ----------
     data_config : DataConfig
         Data configuration.
-    src_files : List[Path]
+    src_files : list[Path]
         List of data files.
-    target_files : Optional[List[Path]], optional
+    target_files : list[Path] or None, optional
         Optional list of target files, by default None.
     read_source_func : Callable, optional
         Read source function for custom types, by default read_tiff.
 
     Attributes
     ----------
-    data_path : List[Path]
+    data_path : list[Path]
         Path to the data, must be a directory.
     axes : str
         Description of axes in format STCZYX.
-    patch_extraction_method : Union[ExtractionStrategies, None]
-        Patch extraction strategy, as defined in extraction_strategy.
-    patch_size : Optional[Union[List[int], Tuple[int]]], optional
+    patch_size : list[int] or tuple[int] or None, optional
         Size of the patches in each dimension, by default None.
-    patch_overlap : Optional[Union[List[int], Tuple[int]]], optional
+    patch_overlap : list[int] or tuple[int] or None, optional
         Overlap of the patches in each dimension, by default None.
-    mean : Optional[float], optional
+    mean : float or None, optional
         Expected mean of the dataset, by default None.
-    std : Optional[float], optional
+    std : float or None, optional
         Expected standard deviation of the dataset, by default None.
-    patch_transform : Optional[Callable], optional
+    patch_transform : Callable or None, optional
         Patch transform callable, by default None.
     """
```

Comment:
Again, I don't think the `"or None"` is necessary for optional params.

Reply:
Also I think the "by default None", can be removed – it is implied by "optional".

Reply:
In this case I do agree since they have a default `None`!